src/components/domain/project/projectLinkModal/ProjectLinkModal.tsx

```diff
@@ -1,141 +1,218 @@
+    alert('연동 성공');
     closeModal();
     setTimeout(() => {
       openModal(<LinkCompleteMessage />);
     });
   };
 
+  const sendCodeMutation = useSendEmailCode(handleSendCodeSuccess);
+  const verifyCodeMutation = useVerifyAuthCode(handleVerifyCodeSuccess);
+  const linkProjectMutation = useLinkProject(handleLinkProjectSuccess);
+
+  // 인증번호 받기 버튼 비활성화 기준 : 선택된 이메일이 없거나, 인증번호를 받은 상태일 때
+  const isDisabledSendButton = !selectedEmail || isCodeSent;
+
+  // 인증하기 버튼 비활성화 기준 : 인증번호가 전송되지 않았거나, 입력창에 값이 없거나, 시간이 끝났을 때
+  const isDisabledVerifyButton = !isCodeSent || !authCode || !timeLeft;
+
+  // 인증번호 받기, 인증하기 form 제출
+  const onSubmit = (data: ProjectLinkForm) => {
+    if (!isCodeSent) {
+      // 인증번호가 전송이 된 상태가 아니라면, 인증번호 전송 api 호출
+      sendCodeMutation.mutate({ email: data.selectedEmail, authType: 'LOAD_PROJECT' });
+    } else {
+      // 인증번호 전송이 진행된 상태라면, 인증하기 api 호출
+      verifyCodeMutation.mutate({
+        email: data.selectedEmail,
+        authCode: data.authCode,
+        authType: 'LOAD_PROJECT',
+      });
+    }
+  };
+
+  const handleSelectMember = (email: string) => {
+    // 인증번호를 받은 상태라면 상태 변경 비활성화
+    if (!isCodeSent) {
+      setValue('selectedEmail', email);
+    }
+  };
+
+  // 해당 프로젝트의 멤버 정보 가져오기
+  const {
+    data,
+    isLoading: memberDataLoading,
+    isError: memberDataError,
+  } = useGetProfileProjectDetails(false, projectId);
+
+  const memberData: MemberData[] = data?.data.members || [];
+  const { sortedMemberData, separatorIndex } = processingMemberData(memberData);
+  const isValidData = !(memberDataLoading || memberDataError || !data);
+
+  // 내가 이미 포함되어 있는 프로젝트라면 return 처리
+  const alertShownRef = useRef(false);
+
+  const isAlreadyMember = sortedMemberData.some((member) => member.userId === loginUser?.userId);
+  if (isAlreadyMember && loginUser?.userId && !alertShownRef.current) {
+    alertShownRef.current = true;
+    closeModal();
+    setTimeout(() => {
+      alert('이미 연동된 프로젝트입니다.');
+    });
+    return null;
+  }
+
   const renderMemberList = () => (
-    <ul className="flex w-full flex-col gap-2">
-      {sortedMemberData.map((member, index) => (
-        <li key={index}>
-          <MemberItem
-            member={member}
-            index={index}
-            isSelected={selectUserEmail === member.email}
-            onSelect={() => handleSelectMember(member.email)}
-          />
-          {index === separatorIndex && <Separator className="my-4" />}
-        </li>
-      ))}
+    <ul className={cn('flex min-h-[300px] w-full flex-col gap-4', !isValidData && 'flex-center')}>
+      {memberDataLoading ? (
+        <ComponentSpinner />
+      ) : memberDataError ? (
+        <span className="text-gray-600 display6">
+          연동할 프로젝트의 팀원을 로드하는 중 오류가 발생했습니다.
+        </span>
+      ) : sortedMemberData.length === 0 ? (
+        <span className="text-gray-600 display6">연동할 프로젝트의 팀원이 없습니다.</span>
+      ) : (
+        sortedMemberData.map((member, index) => (
+          <li key={index}>
```

Comment:
```suggestion
          <li key={member.userId + index}>
```
여기 key값에도 추가하면 불필요한 리렌더링을 줄일 수 있을 것 같습니다! 아래도 `<li key={index}>` 대신 `<li key={role + index}>`을 쓰면 좋을 것 같은데 제 코드에도 그냥 index로 돌린 경우가 많긴 해서,,,언급만 드릴게요

Reply:
sortedMemberData 에서 각 배열을 완벽하게 식별해줄 값이 없어서 (userId에 -1이 여러번 들어갈 수 있음) index를 사용했습니다
제안주신 것 처럼 key값을 써도 실제로 리렌더링 성능에는 큰 차이가 없을 것 같습니다 (여전히 index 값을 포함하니까)
이부분은 객체 정렬할 때 고유 아이디 필드 하나 추가하겠습니다!

Reply:
✅ 8efb1c8eae8b12e10aeb992ca45d5353b7913a94